SECURITY.md

```diff
@@ -1,3 +1,90 @@
-## Reporting a Vulnerability
+# Security Issue Response Process
 
-If you discover a potential security issue in this project we ask that you notify AWS/Amazon Security via our [vulnerability reporting page](http://aws.amazon.com/security/vulnerability-reporting/) or directly via email to aws-security@amazon.com. Please do **not** create a public GitHub issue.
\ No newline at end of file
+## Introduction
+
+Security issues happen as part of the normal lifecycle of software development of OpenSearch. This document describes the process for reporting these issues, responding to these issues, and ensuring we treat them not only with the appropriate priority but also with respect for the discoverers, software providers and their users, and the OpenSearch community in general.
+
+If you discover a potential security issue in this project we ask that you notify the OpenSearch Security Team via email to opensearch-security@amazon.com. Please do **not** create a public GitHub issue.
```

Comment:
regarding opensearch-security@amazon.com:
* how is the email address staffed? is it being monitored 24/7 or only during business hours in a specific location (e.g. US)?
* is this internally a mailing list (i.e. you could eventually add non-AWS employees to the security maintainers) or is it a shared mailbox (i.e. only AWS employees can be added to it)?
* do you support PGP encrypted emails (i'd recommend doing so)? if so, you should list your public key here

Reply:
Although we are hoping to have an @opensearch.org address that could include non-AWS folks, we didn't want that to stop the rolling out of this policy waiting for that to be set up. Currently, the people listed on the security team list on this PR are added to that email address, but no specific commitments as to response times or support are being made at this time.

As for the support of encrypted emails, that's a great idea! Created #91 to track.

Reply:
> but no specific commitments as to response times or support are being made at this time.

maybe no specific commitment is needed, but it'd be good to know whether it can happen that all people with access to the mail address can be on vacation on the same time (think public holiday, etc.) or whether it'd be reasonable to expect a "we got your report and will be looking into it in the next few days and then get back to you" answer within 1-2 business days (thinking of the "we'll update you in 5 business days" emails from AWS security which they seem to be happy to send for several weeks in a row 🙄)

Reply:
Gotcha... I've created #95 to work with the security team and come back with some language around it.